pyccel/parser/semantic.py

```diff
@@ -2049,8 +2050,7 @@ def _visit_For(self, expr, **settings):
         elif isinstance(iterator, list):
             target = []
             for i in iterator:
-                name = str(i.name)
-                var  = Variable('int', name, rank=0)
+                var  = Variable('int', i, rank=0)
```

Comment:
Maybe we should rename `i`->`name` to keep the purpose clear?

Reply:
This can be done directly on the for loop rather than creating a temporary variable